cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,639 @@
+            circuit.append(ops.measure(qubit, key='z'))
+            results = sampler.run(circuit, repetitions=number_shots)
+            excited_probs_l.append(np.mean(results.measurements['z']))
+        gnd_probs.append(1.0 - np.mean(excited_probs_l))
+
+    return RBResult(numbers_cliffords, gnd_probs)
+
+
+def two_qubit_randomized_benchmarking(sampler: sim.SimulatesSamples,
+                                      first_qubit: devices.GridQubit,
+                                      second_qubit: devices.GridQubit,
+                                      numbers_cliffords: Sequence[int],
+                                      number_circuits: int,
+                                      number_shots: int) -> RBResult:
+    """Clifford-based randomized benchmarking (RB) of two qubits.
+
+    A total of number_circuits random circuits are generated, each of which
+    contains a fixed number of two-qubit Clifford gates plus one additional
+    Clifford that inverts the whole sequence and a measurement in the
+    z-basis. Each circuit is repeated number_shots times and the average
+    |00> state population is determined from the measurement outcomes of all
+    of the circuits.
+
+    The above process is done for different numbers of Cliffords specified in
+    numbers_cliffords.
+
+    The two-qubit Cliffords here are decomposed into CZ gates plus single-qubit
+    x and y rotations. See Barends et al., Nature 508, 500 for details.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        first_qubit: The first qubit under test.
+        second_qubit: The second qubit under test.
+        numbers_cliffords: The different numbers of Cliffords in the RB study.
+        number_circuits: The number of random circuits generated for each
+            number of Cliffords.
+        number_shots: The number of repetitions of each circuit.
+
+    Returns:
+        A RBResult object that stores and plots the result.
+    """
+    c1, _, s1, s1_x, s1_y = _single_qubit_cliffords()
+    cfd_matrices = _two_qubit_clifford_matrices(first_qubit, second_qubit,
+                                                c1, s1, s1_x, s1_y)
+    gnd_probs = []
+    for num_cfds in numbers_cliffords:
+        gnd_probs_l = []
+        for _ in range(number_circuits):
+            circuit = _random_two_q_clifford(first_qubit, second_qubit,
+                                             num_cfds, cfd_matrices,
+                                             c1, s1, s1_x, s1_y)
+            circuit.append(ops.measure(first_qubit, second_qubit, key='z'))
+            results = sampler.run(circuit, repetitions=number_shots)
+            gnds = [(not r[0] and not r[1]) for r in results.measurements['z']]
+            gnd_probs_l.append(np.mean(gnds))
+        gnd_probs.append(float(np.mean(gnd_probs_l)))
+
+    return RBResult(numbers_cliffords, gnd_probs)
+
+
+def single_qubit_state_tomography(sampler: sim.SimulatesSamples,
```

Comment:
We can leave these tomography methods for now, but I think they make more sense as "Displays" that can be inserted into circuits. I'm unsure how to best expose the functionality to users. It's likely that someone new who wants to do state tomography will look for a method like this instead of in a display concept they don't know of yet. On the other hand, it's a bit weird for it to be in experiments..

Regardless, it's good to have the code ready to go. It can be refactored into the desired form at a later time.

Reply:
Yeah the display method makes sense. I can refactor the code to do that later on if needs be.